### PR TITLE
fix: Reverse eventsInfo array because of pipeline model change

### PR DESCRIPTION
**Pull request by Code-Beast**
## Context
Events-thumbnail components show outdated events info because of an API bug. Now the API is fixed, we need to make corresponding changes for the UI.

## Objective
Make events-thumbnail show latest events info.

## References
[screwdriver feat(947)](https://github.com/screwdriver-cd/ui/pull/480)

## License
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by adong** [comment]
This would require another small fix. UI needs to pass in the `sort`ing order to be `descending` to `getMetrics`

**Comment by Code-Beast** [comment]
> This would require another small fix. UI needs to pass in the `sort`ing order to be `descending` to `getMetrics`

I checked the data schema and found that pagination schema has a default 'descending', the sort field of query is 'descending' already.

**Comment by Code-Beast** [comment]
> @Code-Beast it's card coded to `ascending`See https://github.com/screwdriver-cd/models/pull/404/files For timeseries it has to be ascending. But we wanted the latest page of data. So now UI has to call pipeline metrics by passing `descending` and then `reverse` it to convert back to time series of latest page of metrics. Also the time information in the card should pertain to the latest event (last entry in time series)

There's one thing I am not sure about. In the query schema of pipeline/metrics, paginate.sort has a default 'descending', thus it makes no difference whether or not we pass a 'sort' query parameter. (In screwdriver, the default sort is 'descending' while in pipeline model, for getMetrics field, the default sort is 'ascending').

File: app/components/collection-view/component.js [hunk 1]
```diff
@@ -171,7 +171,7 @@ export default Component.extend({
           };
 
           if (metrics && metrics.length) {
-            const lastEvent = metrics.get('lastObject');
+            const lastEvent = metrics.get('firstObject');
```
Comment (jithin1987) [inline app/components/collection-view/component.js]: shouldn't this be lastObject itself ? after `reverse` won't  `lastobject`  be the latest event ?

Comment (Code-Beast) [reply app/components/collection-view/component.js]: reverse happens in events-thumbnail. Here in collection-view I am fetching from the original returned metrics, so it should be firstObject

**Comment by jithin1987** [comment]
> There's one thing I am not sure about. In the query schema of pipeline/metrics, paginate.sort has a default 'descending',

@Code-Beast can you point me to where in code this is happening ? As far as I see from model logic, only `metrics` uses `ascending`  by default to generate the time series format.

**Comment by Code-Beast** [comment]
> > There's one thing I am not sure about. In the query schema of pipeline/metrics, paginate.sort has a default 'descending',
> 
> @Code-Beast can you point me to where in code this is happening ? As far as I see from model logic, only `metrics` uses `ascending` by default to generate the time series format.

Sure. https://github.com/screwdriver-cd/data-schema/blob/master/api/pagination.js

**Comment by jithin1987** [comment]
> Sure. https://github.com/screwdriver-cd/data-schema/blob/master/api/pagination.js

@Code-Beast that the base default if it's not definfed. But model is overriding it and had hard coded to `ascending`. See my PRs for relaxing it.

https://github.com/screwdriver-cd/models/pull/404/files
https://github.com/screwdriver-cd/screwdriver/pull/1818/files

**Comment by Code-Beast** [comment]
> > Sure. https://github.com/screwdriver-cd/data-schema/blob/master/api/pagination.js
> 
> @Code-Beast that the base default if it's not defined elsewhere. But model is overriding it and had hard coded to `ascending`. See my PRs for relaxing it.
> 
> https://github.com/screwdriver-cd/models/pull/404/files
> https://github.com/screwdriver-cd/screwdriver/pull/1818/files

image
At this point, 'sort' in the query is the default, which is 'descending', and we pass it the parameter for pipeline.getMetrics();

image
When it is passed in here, it will override the default here, which is 'ascending' and give us the correct  return value for the pagination case.

That's why I think it won't affect the result whether you pass in a 'sort' query parameter when making the API call or not.

**Comment by jithin1987** [comment]
As discussed offline, @Code-Beast  is right abut hapi defaulting to descending. But now it will break Metrics page, since it needs `ascending` sort order. @Code-Beast can you please add that to this PR ?

**Comment by Code-Beast** [comment]
> As discussed offline, @Code-Beast is right abut hapi defaulting to descending. But now it will break Metrics page, since it needs `ascending` sort order. @Code-Beast can you please add that to this PR ?

image

Because only when metrics are fetched by pagination will the 'sort' query parameter be passed in pipeline.getMetrics() within config, current changes won't break metrics page because metrics are not fetched by pagination on that page.

**Pull request merged**